service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/Service.scala

```diff
@@ -263,6 +263,17 @@ object ServiceSupport {
     implicit def methodFor22[Q, R, P1: PPS, P2: PPS, P3: PPS, P4: PPS, P5: PPS, P6: PPS, P7: PPS, P8: PPS, P9: PPS, P10: PPS, P11: PPS, P12: PPS, P13: PPS, P14: PPS, P15: PPS, P16: PPS, P17: PPS, P18: PPS, P19: PPS, P20: PPS, P21: PPS, P22: PPS](f: (P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14, P15, P16, P17, P18, P19, P20, P21, P22) => ServiceCall[Q, R]): ScalaMethodServiceCall[Q, R] = macro methodForImpl22[Q, R]
   }
 
+  private def locateMethod(clazz: Class[_], name: String): Method = {
+    // The class passed in may be an implementation (anonymous, implemented by a macro) of the service, and just doing
+    // a simple getMethod on it may return a method whose declaring class is the implementation, which won't allow us
+    // to invoke it. So we need to search the classes interfaces first for the method, and then if we don't find it
+    // there, we'll search the class itself.
+    val classes = clazz.getInterfaces.toSeq ++ Option(clazz.getSuperclass) :+ clazz
```

Comment:
I think there's still an edge case here. If the superclass contains a concrete implementation of the method, and it is declared in an interface implemented by that superclass, and not directly by `clazz`, then it will still get the concrete implementation.